File: attitude_estimation.py
import numpy as np

## roll fi, pitch theta

def roll_angle(Gpy, Gpz):
    roll = Gpy/Gpz
    fi = np.arctan(roll)*180/np.pi ## converting to arctan and rad to grad
    if(fi > 0):
        fi2 = 180 - fi
    else:
        fi2 = 180 + fi
    #print(Gpy/Gpz, ' -- Gpy/Gpz')
    #print(np.arctan(Gpy/Gpz)*180/np.pi, '--Arctan')
    return round(fi, 4), round(fi2, 4)

def pitch_angle(Gpx, Gpy, Gpz):
    pitch = (-Gpx)/np.sqrt(Gpy**2+Gpz**2) 
    theta = np.arctan(pitch)*180/np.pi ## converting to arctan and rad to grad
    
    #print(theta, ' -- -Gpx/sqrt(Gpy^2+Gpz^2)')
    #print(np.arctan(theta)*180/np.pi, '--Arctan')
    return round(theta, 4)


data = []
attitude = 'attitude_exam.log'

with open(attitude) as f:
    f = f.readlines()

#for line in f:
#    for phrase in line:
#        data.append(line)
#        break

Gpx = 0.461105
Gpy = 0.082198
Gpz = -0.887432
fi, fi2 = roll_angle(Gpy, Gpz)

print(fi, fi2)
print(pitch_angle(Gpx, Gpy, Gpz))

print(data)

File: attitude_estimation.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <cstdlib>
#include <string>
#include <cmath>
#include <ctime>

using namespace std;


double roll_angle(double Gpy, double Gpz) {
    double fi = atan2(Gpy, Gpz)*180/M_PI; // converting to arctan and rad to grad

    return fi;
}

double pitch_angle(double Gpx, double Gpy, double Gpz) {
    double pitch = (-Gpx)/sqrt(pow(Gpy, 2)+pow(Gpz, 2));
    double theta = atan(pitch)*180/M_PI; // converting to arctan and rad to grad

    return theta;
}

int main() {

    vector<string> data;
    vector<vector<string>> organizated;
    
    // open the log
    ifstream file;
    string line;

    file.open("attitude_exam.log"); 
    if (file.is_open()) {
        while (getline(file, line)) {
            data.push_back(line);
        }
        file.close();
    }
    else
    {
        cout << "Unable to open file" << endl;
    }

    for (const auto& element : data) {
        vector<string> temp;
        size_t start = 0;
        size_t end = 0;
        while ((end = element.find(';', start)) != string::npos) {
            temp.push_back(element.substr(start, end-start));
            start = end + 1;
        }
        temp.push_back(element.substr(start));
        organizated.push_back(temp);
    }

    for (const auto& i : organizated) {
        
        string Gpx = i[1];
        string Gpy = i[2];
        string Gpz = i[3];

        cout << "Indíce ---" << endl;
        cout << Gpx << endl;
        cout << Gpy << endl;
        cout << Gpz << endl;
    } 
    
    //double Gpx = 0.461105;
    //double Gpy = 0.082198;
    //double Gpz = -0.1;

    //double fi = roll_angle(Gpy, Gpz);
    //double theta = pitch_angle(Gpx, Gpy, Gpz);

    //cout << fi << endl;
    //cout << thetsa << endl;
    return 0;
}
